src/test/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStepTest.java

```diff
@@ -83,4 +107,117 @@
         }
     }
 
+    @Test
+    public void stepExecutionWithCredentials() throws Exception {
+        assumeNotWindows();
+        
+        IdCredentials registryCredentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "registryCreds", null, "me", "pass");
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), registryCredentials);
+
+        WorkflowJob p = r.createProject(WorkflowJob.class, "prj");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  mockDockerLoginWithEcho {\n" +
+                        "    withDockerRegistry(url: 'https://my-reg:1234', credentialsId: 'registryCreds') {\n" +
+                        "       echo 'config would be set up to connect to https://my-reg:1234'\n" +
```

Comment:
I am not sure what this is proving. You print a literal message, then later assert that the identical message got printed?

Reply:
This `echo` is not needed for the test. I can remove it. The real test here is that the credentials passed to `withDockerRegistry` results in the expected `docker login` command.